packages/block-editor/src/hooks/dimensions.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * WordPress dependencies
+ */
+import { __experimentalProgressiveDisclosurePanel as ProgressiveDisclosurePanel } from '@wordpress/components';
+import { Platform } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { getBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../components/inspector-controls';
+import {
+	MarginEdit,
+	hasMarginSupport,
+	hasMarginValue,
+	resetMargin,
+	useIsMarginDisabled,
+} from './margin';
+import {
+	PaddingEdit,
+	hasPaddingSupport,
+	hasPaddingValue,
+	resetPadding,
+	useIsPaddingDisabled,
+} from './padding';
+
+export const SPACING_SUPPORT_KEY = 'spacing';
+
+/**
+ * Inspector controls for dimensions support.
+ *
+ * @param {Object} props Block props.
+ *
+ * @return {WPElement} Inspector controls for spacing support features.
+ */
+export function DimensionsPanel( props ) {
+	const isPaddingDisabled = useIsPaddingDisabled( props );
+	const isMarginDisabled = useIsMarginDisabled( props );
+	const isDisabled = useIsDimensionsDisabled( props );
+	const isSupported = hasDimensionsSupport( props.name );
+
+	if ( isDisabled || ! isSupported ) {
+		return null;
+	}
+
+	const defaultSpacingControls = getBlockSupport( props.name, [
+		SPACING_SUPPORT_KEY,
+		'__experimentalDefaultControls',
```

Comment:
The addition of another type of flag isn't ideal, I agree. I'm not sure though that I agree with your suggestion, assuming I'm understanding it correctly.

The theme.json allows for a theme developer to completely disable the UI for a particular type of block support feature. If the theme.json disables that UI, the user shouldn't have the ability to turn it on and use it anyway. That undermines the idea of disabling controls via theme.json.

As it stands:

- Block.json --> Allows opt-in for support feature
- Theme.json --> Allows for the UI to be disabled

We need a means for the block to opt in, the theme.json to allow the UI but still hide the less important controls behind the `...` menu.

If not a new flag, could we alter how the theme.json disables controls? The current boolean values to allow/disallow the UI would continue, but it could also support a third option to relegate a control to the `...` menu. I'm not sold on that approach either so very open to fresh ideas.

Reply:
>  If the theme.json disables that UI, the user shouldn't have the ability to turn it on and use it anyway

Yes, what I'm saying is making it impossible to totally disable UI functionality via theme.json it would only hide by default. I think I saw that discussed as a possibility in the past that's why I'm bringing it.

Reply:
Ok, I've missed that discussion around possibly moving away from theme.json having the final say in whether the UI is enabled or not. Do you have any links where I can find if a decision was made in that regard?

Reply:
> Ok, I've missed that discussion around possibly moving away from theme.json having the final say in whether the UI is enabled or not. Do you have any links where I can find if a decision was made in that regard?

No, unfortunately, I just have a vague idea of that option being considered in the past and I don't remember where and by who it was being considered. Maybe @mtias and/or @jasmussen may have better insights and confirm if this idea is a possibility or something we should not follow.